lib/stdlib/test/binary_module_SUITE.erl

```diff
@@ -1373,3 +1376,37 @@ check_no_invalid_read_bug(I) ->
     binary:encode_unsigned(N+N),
     binary:encode_unsigned(N+N, little),
     check_no_invalid_read_bug(I+1).
+
+secure_compare(Config) when is_list(Config) ->
+    true  = binary:secure_compare(<<"">>, <<"">>),
+    true  = binary:secure_compare(<<"good">>, <<"good">>),
+
+    false = binary:secure_compare(<<"good">>, <<"bad">>),
+    false = binary:secure_compare(<<"eh?">>, <<"Hello Joe">>),
+    false = binary:secure_compare(<<"one">>, <<"two">>),
+    false = binary:secure_compare(<<"two">>, <<"one">>),
+    ok.
+
+secure_compare_time(Config) when is_list(Config) -> 
+    A = <<"A">>,
+    B = <<"B">>,
+    
+    %% 9.9 Megabyte base binary
+    Base = binary:copy(<<"X">>, 9999999),
+
+    %% 10 Megabyte arg1 and arg2 with a one byte difference at the start of each. 
+    Arg1 = <<A/binary, Base/binary>>,
+    Arg2 = <<B/binary, Base/binary>>,
+
+    T1 =  erlang:convert_time_unit(erlang:monotonic_time(), native, microsecond),
+
+    binary:secure_compare(Arg1, Arg2),   
+    
+    T2 =  erlang:convert_time_unit(erlang:monotonic_time(), native, microsecond),
+
+    %% Given a 10 megabyte binary we can say with a fair amount of certainty that
```

Comment:
This test and comment might need some more though. On an intel i9 it takes between 12 to 14 milliseconds to complete the op, so I think we can say even if you have a processor that is twice as fast it will still take 6 to 8. I would think that a bif call would never take over 1ms, but we could increase the comparison to `=> 3` if we are concerned about that.